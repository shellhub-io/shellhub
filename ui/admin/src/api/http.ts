import { AxiosInstance } from "axios";
import { setupInterceptorsTo } from "./interceptors";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";

const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("cloud_token") || "";

// eslint-disable-next-line import/no-mutable-exports
let adminApi = new axiosTs.AdminApi(configuration);
// eslint-disable-next-line import/no-mutable-exports
let cloudApi = new axiosTs.CloudApi(configuration);

export const createNewClient = () => {
  const newConfiguration = new Configuration();
  newConfiguration.basePath = `${window.location.protocol}//${window.location.host}`;
  newConfiguration.accessToken = localStorage.getItem("cloud_token") || "";
  adminApi = new axiosTs.AdminApi(newConfiguration);
  cloudApi = new axiosTs.CloudApi(newConfiguration);
  return { adminApi, cloudApi };
};

declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
    getConfiguration(): Configuration | undefined;
    setConfiguration(configuration: Configuration): void;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

/** Returns the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.getConfiguration = function getConfiguration(this: BaseAPI): Configuration | undefined {
  return this.configuration;
};

/** Sets the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.setConfiguration = function setConfiguration(this: BaseAPI, configuration: Configuration): void {
  this.configuration = configuration;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
setupInterceptorsTo(adminApi.axios);

export { adminApi, cloudApi };
