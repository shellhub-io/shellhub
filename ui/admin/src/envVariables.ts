/* eslint-disable import/prefer-default-export */
export const envVariables = {
  announcementsEnable: process.env.SHELLHUB_ANNOUNCEMENTS === "true",
  tinyMceKey: process.env.TINYMCE_API_KEY,
};
