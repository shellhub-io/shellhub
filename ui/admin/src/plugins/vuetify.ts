// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
