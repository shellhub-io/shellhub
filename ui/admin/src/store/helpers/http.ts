import Axios from "axios";
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore //* use while route mapping is not in TypeScript
export default () => {
  const axios = Axios.create({
    baseURL: `${window.location.protocol}//${window.location.host}/admin`,
    headers: {
      Authorization: `Bearer ${localStorage.getItem("cloud_token")}`,
    },
  });

  return axios;
};
