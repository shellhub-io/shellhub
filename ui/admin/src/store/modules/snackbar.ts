import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
