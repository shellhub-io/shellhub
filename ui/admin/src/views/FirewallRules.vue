<template>
  <div>
    <h1>Firewall Rules</h1>
  </div>
  <v-card class="mt-2">
    <FirewallRulesList />
  </v-card>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import FirewallRulesList from "../components/FirewallRules/FirewallRulesList.vue";

export default defineComponent({
  components: { FirewallRulesList },
});
</script>
