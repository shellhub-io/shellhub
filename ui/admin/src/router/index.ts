import { createRouter, createWebHistory, NavigationGuardNext, RouteLocationNormalized } from "vue-router";
import Login from "@admin/views/Login.vue";
import Dashboard from "@admin/views/Dashboard.vue";
import Users from "@admin/views/Users.vue";
import SettingsLicense from "@admin/components/Settings/SettingsLicense.vue";
import SettingsAuthentication from "@admin/components/Settings/SettingsAuthentication.vue";
import Namespaces from "@admin/views/Namespaces.vue";
import Settings from "@admin/views/Settings.vue";
import { computed } from "vue";
import useLicenseStore from "@admin/store/modules/license";
import useLayoutStore from "@admin/store/modules/layout";
import useAuthStore from "@admin/store/modules/auth";
import { plugin as snackbar } from "@/plugins/snackbar"; // using direct plugin because inject() doesn't work outside components

const routes = [
  {
    path: "/login",
    name: "login",
    component: Login,
    meta: {
      layout: "SimpleLayout",
      requiresAuth: false,
    },
  },
  {
    path: "/",
    name: "dashboard",
    component: Dashboard,
  },
  {
    path: "/users",
    name: "users",
    component: Users,
  },
  {
    path: "/user/:id",
    name: "userDetails",
    component: () => import("@admin/views/UserDetails.vue"),
  },
  {
    path: "/devices",
    name: "devices",
    component: () => import("@admin/views/Device.vue"),
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: () => import("@admin/views/Device.vue"),
      },
    ],
  },
  {
    path: "/device/:id",
    name: "deviceDetails",
    component: () => import("@admin/views/DeviceDetails.vue"),
  },
  {
    path: "/sessions",
    name: "sessions",
    component: () => import("@admin/views/Sessions.vue"),
  },
  {
    path: "/session/:id",
    name: "sessionDetails",
    component: () => import("@admin/views/SessionDetails.vue"),
  },
  {
    path: "/settings",
    name: "Settings",
    component: Settings,
    redirect: { name: "SettingProfile" },
    children: [
      {
        path: "authentication",
        name: "SettingAuthentication",
        component: SettingsAuthentication,
      },
      {
        path: "license",
        name: "SettingLicense",
        component: SettingsLicense,
      },
    ],
  },
  {
    path: "/firewall-rules",
    name: "firewall-rules",
    component: () => import("@admin/views/FirewallRules.vue"),
  },
  {
    path: "/firewall-rules/:id",
    name: "firewallRulesDetails",
    component: () => import("@admin/views/FirewallRulesDetails.vue"),
  },
  {
    path: "/namespaces",
    name: "namespaces",
    component: Namespaces,
  },
  {
    path: "/namespace/:id",
    name: "namespaceDetails",
    component: () => import("@admin/views/NamespaceDetails.vue"),
  },
  {
    path: "/announcements",
    name: "announcements",
    component: () => import("@admin/views/Announcements.vue"),
  },
  {
    path: "/announcement/:uuid",
    name: "announcementDetails",
    component: () => import("@admin/views/AnnouncementDetails.vue"),
  },
  {
    path: "/new-announcement",
    name: "new-announcement",
    component: () => import("@admin/views/NewAnnouncement.vue"),
  },
];

const router = createRouter({
  history: createWebHistory("/admin/"),
  routes,
});

router.beforeEach(
  async (to: RouteLocationNormalized, from: RouteLocationNormalized, next: NavigationGuardNext) => {
    const licenseStore = useLicenseStore();
    const layoutStore = useLayoutStore();
    const authStore = useAuthStore();

    const requiresAuth = to.meta.requiresAuth ?? true;

    const layout = to.meta.layout || "AppLayout";
    layoutStore.setLayout(layout as string);

    if (!authStore.isLoggedIn && requiresAuth) {
      return next({
        name: "login",
        query: { redirect: to.fullPath },
      });
    }

    if (authStore.isLoggedIn && !to.meta.requiresAuth) {
      const license = computed(() => licenseStore.getLicense);

      try {
        await licenseStore.get();

        if (license.value.expired && to.name !== "SettingLicense") {
          snackbar.showError("Your license has expired. Please update it and try again.");
          return next({ name: "SettingLicense" });
        }
      } catch {
        if (to.name !== "SettingLicense") {
          snackbar.showError("Failed to get your license info. Please check it and try again.");
          return next({ name: "SettingLicense" });
        }
      }
    }

    return next();
  },
);

export default router;
