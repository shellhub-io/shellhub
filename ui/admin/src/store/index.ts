import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
