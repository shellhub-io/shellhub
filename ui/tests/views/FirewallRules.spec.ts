import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import MockAdapter from "axios-mock-adapter";
import FirewallRules from "@/views/FirewallRules.vue";
import { namespacesApi, usersApi, rulesApi } from "@/api/http";
import { store, key } from "@/store";
import { envVariables } from "@/envVariables";
import { SnackbarPlugin } from "@/plugins/snackbar";
import { router } from "@/router";

type FirewallRulesWrapper = VueWrapper<InstanceType<typeof FirewallRules>>;

describe("Firewall Rules", () => {
  let wrapper: FirewallRulesWrapper;

  const vuetify = createVuetify();

  let mockNamespace: MockAdapter;

  let mockUser: MockAdapter;

  let mockRules: MockAdapter;

  const members = [
    {
      id: "xxxxxxxx",
      username: "test",
      role: "owner",
    },
  ];

  const namespaceData = {
    name: "test",
    owner: "test",
    tenant_id: "fake-tenant-data",
    members,
    settings: {
      session_record: true,
    },
    max_devices: 3,
    devices_count: 3,
    created_at: "",
  };

  const authData = {
    status: "success",
    token: "",
    user: "test",
    name: "test",
    tenant: "fake-tenant-data",
    email: "test@test.com",
    id: "xxxxxxxx",
    role: "owner",
    mfa: {
      enable: false,
      validate: false,
    },
  };

  const res = {
    data: [namespaceData],
    headers: {
      "x-total-count": 1,
    },
  };

  const firewallRule = {
    data: [{
      priority: 1,
      action: "allow",
      active: true,
      filter: {
        hostname: ".*",
      },
      source_ip: ".*",
      username: ".*",
    }],
    headers: {
      "x-total-count": 1,
    },
  };

  beforeEach(async () => {
    vi.useFakeTimers();
    localStorage.setItem("tenant", "fake-tenant-data");

    envVariables.isCloud = true;

    mockNamespace = new MockAdapter(namespacesApi.getAxios());
    mockUser = new MockAdapter(usersApi.getAxios());
    mockRules = new MockAdapter(rulesApi.getAxios());

    mockRules.onGet("http://localhost:3000/api/firewall/rules?page=1&per_page=10").reply(200, firewallRule.data);
    mockNamespace.onGet("http://localhost:3000/api/namespaces/fake-tenant-data").reply(200, namespaceData);
    mockUser.onGet("http://localhost:3000/api/auth/user").reply(200, authData);

    store.commit("auth/authSuccess", authData);
    store.commit("auth/changeData", authData);
    store.commit("namespaces/setNamespace", namespaceData);
    store.commit("namespaces/setNamespaces", res);
    store.commit("firewallRules/setFirewalls", firewallRule);
    store.commit("firewallRules/setFirewall", firewallRule.data[0]);

    wrapper = mount(FirewallRules, {
      global: {
        plugins: [[store, key], vuetify, router, SnackbarPlugin],
        config: {
          errorHandler: () => { /* ignore global error handler */ },
        },
      },
    });
  });

  afterEach(() => {
    vi.useRealTimers();
    vi.restoreAllMocks();
    wrapper.unmount();
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders the template with data", () => {
    expect(wrapper.find('[data-test="firewall-rules"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="help-icon"]').exists()).toBe(true);
  });

  it("Toggles showHelp when help icon is clicked", async () => {
    const helpIcon = wrapper.find('[data-test="help-icon"]');
    await helpIcon.trigger("click");
    expect(wrapper.vm.showHelp).toBe(true);

    await helpIcon.trigger("click");
    expect(wrapper.vm.showHelp).toBe(false);
  });
});
