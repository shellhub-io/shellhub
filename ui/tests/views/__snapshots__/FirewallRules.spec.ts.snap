// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Firewall Rules > Renders the component 1`] = `
"<div class="d-flex flex-column justify-space-between align-center flex-sm-row mb-2" data-test="firewall-rules">
  <div class="d-flex align-center">
    <h1>Firewall Rules</h1><i class="mdi-help-circle mdi v-icon notranslate v-theme--light v-icon--size-small v-icon--clickable ml-2" role="button" aria-hidden="false" tabindex="0" data-test="help-icon"></i>
    <div class="v-spacer"></div>
    <div class="v-spacer"></div>
  </div>
  <div>
    <div aria-describedby="v-tooltip-v-0"><button type="button" class="v-btn v-btn--elevated v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated" tabindex="0" aria-label="Dialog Add device" data-test="firewall-add-rule-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""> Add Rule </span>
        <!---->
        <!---->
      </button></div>
    <!--teleport start-->
    <!--teleport end-->
    <!---->
    <!---->
  </div>
</div>
<!--v-if-->
<div>
  <div>
    <div data-v-046d4556="" data-test="firewallRules-list">
      <div data-v-046d4556="" class="v-table v-theme--light v-table--density-default bg-background border rounded">
        <!---->
        <div class="v-table__wrapper">
          <table>
            <thead data-v-046d4556="" class="bg-v-theme-background">
              <tr data-v-046d4556="">
                <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Active</span></th>
                <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Priority</span></th>
                <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Action</span></th>
                <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Source IP</span></th>
                <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Username</span></th>
                <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Filter</span></th>
                <th data-v-046d4556="" class="text-center"><span data-v-046d4556="">Actions</span></th>
              </tr>
            </thead>
            <tbody data-v-046d4556="">
              <tr>
                <td class="text-center"><i class="mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-default text-success" aria-hidden="true" data-test="firewall-rules-active"></i></td>
                <td class="text-center" data-test="firewall-rules-priority">1</td>
                <td class="text-center" data-test="firewall-rules-action">Allow</td>
                <td class="text-center" data-test="firewall-rules-source-ip">Any IP</td>
                <td class="text-center" data-test="firewall-rules-username">All users</td>
                <td class="text-center" data-test="firewall-rules-filter">
                  <div>All devices</div>
                </td>
                <td class="text-center"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-comfortable v-btn--size-default v-btn--variant-plain border rounded bg-v-theme-background" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-2" data-test="firewall-rules-actions"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                    <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-format-list-bulleted mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                    <!---->
                    <!---->
                  </button>
                  <!--teleport start-->
                  <!--teleport end-->
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <!---->
      </div>
      <!--v-if-->
      <div data-v-046d4556="" class="d-flex w-100 justify-end align-center"><span data-v-046d4556="" class="text-subtitle-2 mr-4">Items per page:</span>
        <div data-v-046d4556="">
          <div data-v-046d4556="" class="v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-input--plain-underlined v-combobox v-combobox--single mb-4">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--active v-field--appended v-field--dirty v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-8">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!---->
                  <!---->
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!---->
                    <div class="v-combobox__selection"><span class="v-combobox__selection-text">10<!----></span></div><input size="1" type="text" id="input-v-6" aria-describedby="input-v-6-messages" outlined="" value="10">
                  </div>
                  <!---->
                </div>
                <!---->
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-combobox__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
        </div>
        <div data-v-046d4556="" class="d-flex align-center"><button data-v-046d4556="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button><span data-v-046d4556="" class="text-subtitle-2">1 of 1</span><button data-v-046d4556="" type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
            <!---->
            <!---->
          </button></div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;
